src/Features/Core/Portable/Rename/SymbolicRenameInfo.cs

```diff
@@ -0,0 +1,259 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.LanguageService;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    internal sealed class SymbolicRenameInfo
+    {
+        private const string AttributeSuffix = "Attribute";
+
+        [MemberNotNullWhen(true, nameof(LocalizedErrorMessage))]
+        [MemberNotNullWhen(false, nameof(Document))]
+        [MemberNotNullWhen(false, nameof(TriggerText))]
+        [MemberNotNullWhen(false, nameof(Symbol))]
+        public bool IsError => LocalizedErrorMessage != null;
+
+        public string? LocalizedErrorMessage { get; }
+
+        public Document? Document { get; }
+        public SyntaxToken TriggerToken { get; }
+        public string? TriggerText { get; }
+        public ISymbol? Symbol { get; }
+        public bool ForceRenameOverloads { get; }
+        public ImmutableArray<DocumentSpan> DocumentSpans { get; }
+
+        public bool IsRenamingAttributePrefix { get; }
+
+        private SymbolicRenameInfo(string localizedErrorMessage)
+            => this.LocalizedErrorMessage = localizedErrorMessage;
+
+        private SymbolicRenameInfo(
+            Document document,
+            SyntaxToken triggerToken,
+            string triggerText,
+            ISymbol symbol,
+            bool forceRenameOverloads,
+            ImmutableArray<DocumentSpan> documentSpans)
+        {
+            Document = document;
+            TriggerToken = triggerToken;
+            TriggerText = triggerText;
+            Symbol = symbol;
+            ForceRenameOverloads = forceRenameOverloads;
+            DocumentSpans = documentSpans;
+
+            this.IsRenamingAttributePrefix = CanRenameAttributePrefix(triggerText);
+        }
+
+        private bool CanRenameAttributePrefix(string triggerText)
+        {
+            Contract.ThrowIfTrue(this.IsError);
+
+            // if this isn't an attribute, or it doesn't have the 'Attribute' suffix, then clearly
+            // we can't rename just the attribute prefix.
+            if (!IsRenamingAttributeTypeWithAttributeSuffix())
+                return false;
+
+            // Ok, the symbol is good.  Now, make sure that the trigger text starts with the prefix
+            // of the attribute.  If it does, then we can rename just the attribute prefix (otherwise
+            // we need to rename the entire attribute).
+#pragma warning disable IDE0059 // Unnecessary assignment of a value - https://github.com/dotnet/roslyn/issues/45895
+            var nameWithoutAttribute = GetWithoutAttributeSuffix(this.Symbol.Name);
+
+            return triggerText.StartsWith(triggerText); // TODO: Always true? What was it supposed to do?
+#pragma warning restore IDE0059 // Unnecessary assignment of a value
```

Comment:
note: i copied this code from teh original.  but afaict it's just super busted.  teh call to GetWithoutAttributeSuffix is just completely ignored.  likely why we have existing bugs on renaming attributes not working properly.